### PR TITLE
Remove labels overwrite functionality

### PR DESCRIPTION
**Pull request by lcarva**

### CODE REVIEW

**Comment by twaugh** [comment]
This removes the `labels` parameter that koji-containerbuild is currently using, so applying this change on a Koji builder would result in failures.

Don't we need to adjust koji-containerbuild first, or else accept the `labels` parameter and ignore it?

**Comment by lcarva** [comment]
I'll do some additional testing, but k-cb uses the create_build API method
which takes arbitrary keyword arguments. It should simply ignore unknown
ones. Also added a unit test to verify this behavior.

**Comment by twaugh** [comment]
Oh, good point.

File: tests/test_api.py [hunk 1]
```diff
@@ -68,6 +68,34 @@ def test_get_pod_for_build(self, osbs):
...
+            'git_branch': TEST_GIT_BRANCH,
+            'user': TEST_USER,
+            'component': TEST_COMPONENT,
+            'target': TEST_TARGET,
+            'architecture': TEST_ARCH,
+            'yum_repourls': None,
+            'koji_task_id': None,
+            'scratch': False,
+            # Stuff that should be ignored and not cause erros
+            'labels': {'Release': 'bacon'},
```
Comment (twaugh) [inline tests/test_api.py]: OK, I didn't spot this before.

**Comment by twaugh** [comment]
LGTM

**Pull request merged**